### PR TITLE
Todo features

### PR DESCRIPTION
**Pull request by helciodev**
The features of this project are, user can:

1. Create lists of todo.
2. Add tasks to each list.
3. Delete and edit tasks
4. Delete lists

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+const taskCountElement = document.getElementById('task-count');
+let selectedList = lists[lists.length - 1] || '';
+const modalParent = document.getElementById('modal-parent');
+
+// clear element function
+function clearElement(element) {
+  while (element.firstChild) {
+    element.removeChild(element.firstChild);
+  }
+}
```
Comment (SajjadAhmad14) [inline src/index.js]: Please use the arrow function here,

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+function createTask(name, priority, dueDate, description) {
+  return {
+    id: Date.now().toString(), name, complete: false, description, dueDate, priority,
+  };
+}
+
+function save() {
+  localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(lists));
+  localStorage.setItem(LOCAL_SELECTED_LIST, JSON.stringify(selectedList));
+}
```
Comment (SajjadAhmad14) [inline src/index.js]: Please use the arrow function here,

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+function priorityTaskColor(priorityTask) {
+  if (priorityTask.textContent === 'low') {
+    priorityTask.classList.add('text-gray-500', 'font-bold');
+  } else if (priorityTask.textContent === 'medium') {
+    priorityTask.classList.add('text-yellow-500', 'font-bold');
+  } else {
+    priorityTask.classList.add('text-red-500', 'font-bold');
+  }
+}
```
Comment (SajjadAhmad14) [inline src/index.js]: Please use the arrow function here,

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    checkbox.classList.add('checked');
+    checkbox.checked = task.complete;
+    const edit = templateElement.getElementById('edit');
+    const deleteTask = templateElement.getElementById('delete');
+    [checkbox, edit, deleteTask].forEach((templateChild) => {
+      templateChild.dataset.taskIndex = taskIndex;
+    });
+    tasksList.appendChild(templateElement);
+  });
+}
```
Comment (SajjadAhmad14) [inline src/index.js]: Please use the arrow function here,

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+function tasksCount() {
+  const tasksToComplete = selectedList.tasks.filter((task) => !task.complete === true).length;
+
+  if (tasksToComplete === 1) {
+    return `${tasksToComplete} task remaining`;
+  }
+  return `${tasksToComplete} tasks remaining`;
+}
```
Comment (SajjadAhmad14) [inline src/index.js]: Please use the arrow function here,

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+  }
+
+  if (descriptionModal.value !== null || descriptionModal.value !== '') {
+    taskToEdit.description = descriptionModal.value;
+  }
+
+  if (dueDateModal.value !== null || dueDateModal.value !== '') {
+    taskToEdit.dueDate = dueDateModal.value;
+  }
+}
```
Comment (SajjadAhmad14) [inline src/index.js]: Please use the arrow function here,

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+
+function renderCurrentValues(currentListTask) {
+  taskInputModal.value = currentListTask.name;
+  dueDateModal.value = currentListTask.dueDate;
+  optionsModal.value = currentListTask.priority;
+  descriptionModal.value = currentListTask.description;
+}
```
Comment (SajjadAhmad14) [inline src/index.js]: Please use the arrow function here,

File: src/index.js [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+    renderList();
+    taskRenderFn();
+    taskCountElement.textContent = tasksCount();
+  } else if (event.target.id === 'edit' || event.target.id === 'close-btn') {
+    modalParent.classList.toggle('center');
+    renderCurrentValues(selectedList.tasks[event.target.dataset.taskIndex]);
+    saveBtnModal.addEventListener('click', () => {
+      editTasks(selectedList.tasks[event.target.dataset.taskIndex]);
+      modalParent.classList.toggle('center');
+      taskRenderFn();
```
Comment (SajjadAhmad14) [inline src/index.js]: DOM manipulation and app logic should be completely separate.

File: src/index.js [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+  selectedList.tasks.push(newTask);
+  save();
+  taskRenderFn();
+  taskCountElement.textContent = tasksCount();
+});
+
+listContainer.addEventListener('click', (event) => {
+  lists = lists.filter((list) => list.id !== event.target.dataset.element);
+  saveAndRender();
+});
```
Comment (SajjadAhmad14) [inline src/index.js]: DOM manipulation and app logic should be completely separate.

File: src/index.js [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+  selectedList = lists.find((list) => list.id === event.target.dataset.listId);
+  save();
+
+  if (event.target.tagName.toLowerCase() === 'li') {
+    tasks.classList.remove('hidden');
+    listTitle.textContent = event.target.textContent;
+    taskCountElement.textContent = tasksCount();
+  }
+  taskRenderFn();
+});
```
Comment (SajjadAhmad14) [inline src/index.js]: DOM manipulation and app logic should be completely separate.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [x] You have mastered Git flow.
- [x] README is professional contains all the info to run the project locally.
- [x] Your app UI is pleasing.
## To Improve 🙏
- [x] Please see inline comments for reference.
- [x] You have all your code in one file. Please separate your code into different modules. It's easy to manage code and debug in modules.
- [x] DOM-Manipulation and Logic are in separate modules. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/modules_requirement.png).
- [x] There should be one default project.
- [x] Please use ES6 syntax. Use arrow functions instead of function keywords.

### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Review by Berabjesus — APPROVED** [review]
# **Status: PROJECT APPROVED** 
Hey @helciodev great job implementing the changes required by the previous reviewer.
**All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete. 👍** 

### **Before you merge to the default branch please have a look at the following OPTIONAL comments below.**

- [ ] [OPTIONAL] The MVC way of approach you have used here means that the model section of your code should be responsible for data access, in this case handling local storage. Currently, the model section looks more like a helper than a model. Please have a look into that.

I wish you all the very best in your next tasks:1st_place_medal:
#### **If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help** 
**HAPPY CODING :computer:[Bereket Beshane](https://bereketababu.me)** | [LinkedIn](https://linkedin.com/in/bereketbeshane) | [GitHub](https://github.com/Berabjesus)

**Pull request merged**